### PR TITLE
Relay control

### PR DESCRIPTION
**Pull request by StWiemann**
Added support for I2C base I/O extension. 7 Relays can now be controlled and their state can be read using a PCF8574 (RelayControl Class). Also I/O input can now be read through another PCF8574 (I2CInput class). I2C-addresses of the PCF8574 is hard coded.

### CODE REVIEW

File: vosekast_control/I2cInputAndRelayControl.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import smbus
+from typing import Dict, List
+
+
+class RelayControl:
+    def __init__(self):
+        self.address_relays = 0x38
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:7]: What, if we want to use a second relay with a other address. Maybe you can move that to the params with a default value.
And maybe you should only say `address` and not `address_relays`.

File: vosekast_control/I2cInputAndRelayControl.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        self.bus = smbus.SMBus(1)
+        self.state_dict = {}
+        self.state_reading = None
+        self.binary = 255  # Represents relay address and inverted state in binary e.g. 0b11110101 -> relay 2 and 4 are "on"
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:11]: binary as name is confusing. called `state`. maybe `binary_state`.

File: vosekast_control/I2cInputAndRelayControl.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import smbus
+from typing import Dict, List
+
+
+class RelayControl:
+    def __init__(self):
+        self.address_relays = 0x38
+        self.bus = smbus.SMBus(1)
+        self.state_dict = {}
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:9]: you use the `state_dict` only in a getter function without caching. So to store that variable in the class context makes no sense.

File: vosekast_control/I2cInputAndRelayControl.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        self._flash()
+
+    def get_state_dict(self) -> Dict[int, int]:
+        bin_state = format(self.binary, "07b")
+        n = len(bin_state)
+        for digit in bin_state:
+            self.state_dict[n] = 1 ^ int(digit)
+            n -= 1
+
+        return self.state_dict
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:34]:
```suggestion
        overview = {}
        
        for i in range(8):
            overview[i] = (self.binary >> i) & 1
            
        return overview
```

File: vosekast_control/I2cInputAndRelayControl.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    def all_on(self):
+        self.binary = 0
+        self._flash()
+
+    def _flash(self):
+        self.bus.write_byte_data(self.address_relays, 0, self.binary)
+
+
+class I2CInput:
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:53]: The first you called `RelayControl` and now `I2CInput`?

File: vosekast_control/I2cInputAndRelayControl.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import smbus
+from typing import Dict, List
+
+
+class RelayControl:
+    def __init__(self):
+        self.address_relays = 0x38
+        self.bus = smbus.SMBus(1)
+        self.state_dict = {}
+        self.state_reading = None
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:10]: what is the difference between `state_reading` and `binary`?

File: vosekast_control/I2cInputAndRelayControl.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+    def _flash(self):
+        self.bus.write_byte_data(self.address_relays, 0, self.binary)
+
+
+class I2CInput:
+    def __init__(self):
+        self.address_input = 0x39
+        self.bus = smbus.SMBus(1)
+        self.state_reading = None
```
Comment (karlTGA) [inline vosekast_control/I2cInputAndRelayControl.py:57]: You only write that variable and never read the value in a other place. Maybe you can drop that property.

File: vosekast_control/I2cInputAndRelayControl.py (hunk 6 above)
Comment (StWiemann) [reply vosekast_control/I2cInputAndRelayControl.py:10]: `state_reading` is the state as read from i2c. `binary` is the current state of the relays as stored in the property. Might differ when restarted. And is hence not the same.

**Pull request merged**